src/utils/__tests__/saml.test.ts

```diff
@@ -54,6 +59,43 @@ describe("getXmlFromSamlResponse", () => {
   });
 });
 
+describe("getAuthorizeRequestTamperer", () => {
+  it("should add professional spid extension", async () => {
+    await getAuthorizeRequestTamperer(
+      // spid-testenv does not accept an xml header with utf8 encoding
+      new Builder({ xmldec: { encoding: undefined, version: "1.0" } }),
+      {
+        professionalSpidExtension: {
+          professionalSpidEnabled: true,
+          purpose: "P"
+        }
+        // tslint:disable-next-line: no-any
+      } as any,
+      samlConfig
+    )(samlRequest)
+      .fold(
+        err => fail(err),
+        _ => expect(_).toContain("<spid:Purpose>P</spid:Purpose>")
+      )
+      .run();
+  });
+
+  it("should not add professional spid extension", async () => {
+    await getAuthorizeRequestTamperer(
+      // spid-testenv does not accept an xml header with utf8 encoding
+      new Builder({ xmldec: { encoding: undefined, version: "1.0" } }),
+      // tslint:disable-next-line: no-any
+      {} as any,
+      samlConfig
+    )(samlRequest)
+      .fold(
+        err => fail(err),
+        _ => expect(_).not.toContain("<spid:Purpose>P</spid:Purpose>")
```

Comment:
I'd test on a less restrictive pattern, like `<spid:Purpose`, does it have sense?